server/block/mushroom_type.go

```diff
@@ -0,0 +1,28 @@
+package block
+
+// MushroomType represents a type of mushroom. This is either brown or red.
+type MushroomType struct {
+	mushroom
+}
+
+// BrownMushroom returns the brown mushroom variant of mushrooms.
+func BrownMushroom() MushroomType {
+	return MushroomType{mushroom(0)}
+}
+
+// RedMushroom returns the red mushroom variant of mushrooms.
+func RedMushroom() MushroomType {
+	return MushroomType{mushroom(1)}
+}
+
+// MushroomTypes returns a list of all mushroom types.
+func MushroomTypes() []MushroomType {
+	return []MushroomType{BrownMushroom(), RedMushroom()}
+}
+
+type mushroom uint8
+
+// Uint8 returns the mushroom as an uint8.
+func (m mushroom) Uint8() uint8 {
```

Comment:
Could maybe replace this with a `String` method instead and return "red" or "brown" so you can use it in EncodeBlock and EncodeItem.